test/impl-base/src/main/java/org/jboss/arquillian/test/impl/EventTestRunnerAdaptor.java

```diff
@@ -48,6 +51,8 @@
 {
    private Manager manager;
    
+   private static final ThreadLocal<TestExecutionDecider> testExecutionDecider = new ThreadLocal<TestExecutionDecider>();
```

Comment:
Avoid any caching here. Resolve it from the ServiceLoader each time, let it cache if required. The service should use the contexts etc if it require other resources via injection.